datasets/ar_sarcasm/ar_sarcasm.py

```diff
@@ -0,0 +1,110 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+
+import datasets
+
+
+# no BibTeX citation
+_CITATION = """@inproceedings{abu-farha-magdy-2020-arabic,
+    title = "From {A}rabic Sentiment Analysis to Sarcasm Detection: The {A}r{S}arcasm Dataset",
+    author = "Abu Farha, Ibrahim  and Magdy, Walid",
+    booktitle = "Proceedings of the 4th Workshop on Open-Source Arabic Corpora and Processing Tools, with a Shared Task on Offensive Language Detection",
+    month = may,
+    year = "2020",
+    address = "Marseille, France",
+    publisher = "European Language Resource Association",
+    url = "https://www.aclweb.org/anthology/2020.osact-1.5",
+    pages = "32--39",
+    language = "English",
+    ISBN = "979-10-95546-51-1",
+}"""
+
+_DESCRIPTION = """\
+ArSarcasm is a new Arabic sarcasm detection dataset.
+The dataset was created using previously available Arabic sentiment analysis datasets (SemEval 2017 and ASTD)
+ and adds sarcasm and dialect labels to them. The dataset contains 10,547 tweets, 1,682 (16%) of which are sarcastic.
+"""
+
+_LICENSE = "MIT"
+
+_URLs = {
+    "default": "https://github.com/iabufarha/ArSarcasm/archive/master.zip",
+}
+
+
+class ArSarcasm(datasets.GeneratorBasedBuilder):
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "dialect": datasets.Value("string"),
+                "sarcasm": datasets.Value("bool"),
```

Comment:
Can you make it a ClassLabel feature ?
```python
"sarcasm": datasets.ClassLabel(names=["non-sarcastic", "sarcastic"]),
```

Also if you think that other fields (like `dialect` or `sentiment`) might be used as label references you can do the same for them.

You'll have to regenerate the dataset_infos.json after this change.